src/vs/editor/contrib/codeAction/browser/codeActionController.ts

```diff
@@ -137,8 +142,17 @@ export class CodeActionController extends Disposable implements IEditorContribut
 		}
 
 		MessageController.get(this._editor)?.closeMessage();
-		const triggerPosition = this._editor.getPosition();
-		this._trigger({ type: CodeActionTriggerType.Invoke, triggerAction, filter, autoApply, context: { notAvailableMessage, position: triggerPosition } });
+
+		if (this.recentlyChanged) {
+			setTimeout(() => {
```

Comment:
This approach seems a little scary since it will effect all code actions. Can this live in the TS extension instead to reduce the impact? The idea:

1. VS code makes the request code action right away
2. TS extension knows that it may not have up to date diagnostics so it delays sending the request to TS Server. 
3. After the delay, it figures out the latest diagnostics and uses these to request against TS Server